name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/variableurl/Factory.java

```diff
@@ -1,13 +1,40 @@
 package name.abuchen.portfolio.online.impl.variableurl;
 
+import name.abuchen.portfolio.online.impl.variableurl.macros.ConstString;
+import name.abuchen.portfolio.online.impl.variableurl.macros.Currency;
+import name.abuchen.portfolio.online.impl.variableurl.macros.FormattedDate;
+import name.abuchen.portfolio.online.impl.variableurl.macros.ISIN;
+import name.abuchen.portfolio.online.impl.variableurl.macros.Macro;
+import name.abuchen.portfolio.online.impl.variableurl.macros.PageNumber;
+import name.abuchen.portfolio.online.impl.variableurl.macros.TickerSymbol;
+import name.abuchen.portfolio.online.impl.variableurl.macros.WKN;
+import name.abuchen.portfolio.online.impl.variableurl.urls.ConstURL;
+import name.abuchen.portfolio.online.impl.variableurl.urls.VariableURL;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class Factory
 {
-    private static final Pattern URL_MACRO_ABSTRACT = Pattern.compile("\\{(.*?)}"); //$NON-NLS-1$
+    private interface MacroParser
+    {
+        Macro parse(CharSequence input) throws IllegalArgumentException;
+    }
+
+    private static final Pattern URL_MACRO_ABSTRACT = Pattern.compile("\\{(.*?)}");
```

Comment:
Nicht zu übersetzende Textzeichen werden bei Eclipse mit einem ```//$NON-NLS-1$``` versehen. Siehe auch in dem ursprünglichen Code. Ich versuche nach Möglichkeit *keine* Warnungen im Code zu haben. In Eclipse kannst du das einfach per *Ctrl-.* mit dem entsprechenden Kommentar versehen. Nicht super wichtig - aber es macht mir das Leben einfacher wenn Du gerade drauf achten würdest.

Reply:
> Gilt das auch für Unit Tests?

In den Tests mache ich das nicht für jeden String, sondern füge oben einfach die passende Annotation an der Klasse hinzu: ```@SuppressWarnings("nls")```